Tutorials/src/eventMixing.cxx

```diff
@@ -32,112 +35,316 @@ DECLARE_SOA_TABLE(Hashes, "AOD", "HASH", hash::Bin);
 
 } // namespace o2::aod
 
-struct MixedEventsPartitionedTracks {
+struct MixedEvents {
   std::vector<double> xBins{VARIABLE_WIDTH, -0.064, -0.062, -0.060, 0.066, 0.068, 0.070, 0.072};
   std::vector<double> yBins{VARIABLE_WIDTH, -0.320, -0.301, -0.300, 0.330, 0.340, 0.350, 0.360};
-  BinningPolicy<aod::collision::PosX, aod::collision::PosY> binningOnPositions{{xBins, yBins}, true}; // true is for 'ignore overflows' (true by default)
+  using BinningType = BinningPolicy<aod::collision::PosX, aod::collision::PosY>;
+  BinningType binningOnPositions{{xBins, yBins}, true}; // true is for 'ignore overflows' (true by default)
+  SameKindPair<aod::Collisions, aod::Tracks, BinningType> pair{binningOnPositions, 5, -1};
 
-  Filter trackFilter = (aod::track::x > -0.8f) && (aod::track::x < 0.8f) && (aod::track::y > 1.0f);
-  using myTracks = soa::Filtered<aod::Tracks>;
+  void process(aod::Collisions const& collisions, aod::Tracks const& tracks)
+  {
+    LOGF(info, "Input data Collisions %d, Tracks %d ", collisions.size(), tracks.size());
 
-  Configurable<float> philow{"phiLow", 1.0f, "lowest phi"};
-  Configurable<float> phiup{"phiUp", 2.0f, "highest phi"};
+    int count = 0;
+    for (auto& [c1, tracks1, c2, tracks2] : pair) {
+      int bin = binningOnPositions.getBin({c1.posX(), c1.posY()});
+      LOGF(info, "Mixed event bin: %d collisions: (%d, %d) / %d, tracks: (%d, %d) / %d", bin, c1.globalIndex(), c2.globalIndex(), collisions.size(), tracks1.size(), tracks2.size(), tracks.size());
```

Comment:
I would focus here on what the user needs (so remove bin (see above) and collisions and tracks)